<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <param name="robot_description" command="$(find xacro)/xacro $(find tams_lwr_wsg50_description)/urdf/lwr_with_wsg50.xacro" />
  <arg name="control_mode" default="10" />
  <rosparam>
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 1000
    tool_force_torque_controller:
      type: force_torque_sensor_controller/ForceTorqueSensorController
      publish_rate: 1000
    lwr:
      type: position_controllers/JointTrajectoryController
      joints:
        - lwr_arm_0_joint
        - lwr_arm_1_joint
        - lwr_arm_2_joint
        - lwr_arm_3_joint
        - lwr_arm_4_joint
        - lwr_arm_5_joint
        - lwr_arm_6_joint
      joint_trajectory_action_node:
        joints:
          - lwr_arm_0_joint
          - lwr_arm_1_joint
          - lwr_arm_2_joint
          - lwr_arm_3_joint
          - lwr_arm_4_joint
          - lwr_arm_5_joint
          - lwr_arm_6_joint
  </rosparam>
  <node name="ros_fri_roscontrol" pkg="ros_fri" type="ros_fri_roscontrol" respawn="false" output="screen" required="true">
    <param name="control_mode" value="$(arg control_mode)" />
    <remap from="/joint_states" to="/lwr/joint_states" />
  </node>
  <node name="position_trajectory_controller" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="lwr tool_force_torque_controller joint_state_controller" required="true"/>
</launch>
